streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java

```diff
@@ -154,4 +155,51 @@ public static void purgeLocalStreamsState(Properties streamsConfiguration) throw
         produceKeyValuesSynchronously(topic, keyedRecords, producerConfig);
     }
 
+    /**
+     * Wait until enough data (key-value records) has been consumed
+     * @param consumerConfig Kafka Consumer configuration
+     * @param topic          Topic to consume grom
+     * @param expectedNumRecords Minimum number of expected records
+     * @param waitTime       Upper bound in waiting time
+     * @return All the records consumed, or null if no records are consumed
+     * @throws InterruptedException
+     */
+    public static <K, V> List<KeyValue<K, V>> waitUntilKeyValuesMatch(Properties consumerConfig, String topic,
+                                                                     int expectedNumRecords,
+                                                                     long waitTime) throws InterruptedException {
+        List<KeyValue<K, V>> readData;
+        long startTime = System.currentTimeMillis();
+        while (true) {
+            readData = readKeyValues(topic, consumerConfig);
+            if (readData.size() >= expectedNumRecords)
+                return readData;
+            if (System.currentTimeMillis() > startTime + waitTime)
+                return null;
+            Thread.sleep(Math.min(waitTime, 100L));
+        }
+    }
+
+    /**
+     * Wait until enough data (value records) has been consumed
+     * @param consumerConfig Kafka Consumer configuration
+     * @param topic          Topic to consume grom
+     * @param expectedNumRecords Minimum number of expected records
+     * @param waitTime       Upper bound in waiting time
+     * @return All the records consumed, or null if no records are consumed
+     * @throws InterruptedException
+     */
+    public static <V> List<V> waitUntilValuesMatch(Properties consumerConfig, String topic,
+                                                   int expectedNumRecords, int maxMessages,
+                                                   long waitTime) throws InterruptedException {
+        List<V> readData;
+        long startTime = System.currentTimeMillis();
+        while (true) {
+            readData = readValues(topic, consumerConfig, maxMessages);
+            if (readData.size() >= expectedNumRecords)
+                return readData;
+            if (System.currentTimeMillis() > startTime + waitTime)
+                return null;
+            Thread.sleep(Math.min(waitTime, 100L));
+        }
```

Comment:
Thanks @ijuma . I addressed all comments, but I'll skip this one since it's not worth generalizing for what ultimately is a loop around sleep.